test/new-e2e/utils/e2e/e2e_test.go

```diff
@@ -7,35 +7,50 @@ import (
 	ec2vm "github.com/DataDog/test-infra-definitions/aws/scenarios/vm/ec2VM"
 	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
 	commonos "github.com/DataDog/test-infra-definitions/common/os"
+	"github.com/DataDog/test-infra-definitions/datadog/agent"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 )
 
 type MyEnv struct {
-	VM *client.VM
+	VM    *client.VM
+	Agent *client.Agent
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Might require some comments, as per why we pass both the vm and the agent running on it. Just an example, ad here tests help.

Reply:
Isn't the example not clear enough as it uses both the VM and the Agent?

Reply:
Yes test helps, we might give more details in the doc